### DISCUSSION TITLE
How to suggest and help get a new device added to database

### DISCUSSION THREAD

**Discussion by Anto79-ops** [zwave-js/zwave-js#4902] [General] [answered]
Hello!

I have this product, the alarm.com (ADC) SWM150, made by Building 36.

Its not only a open-close type water valve but it also does metering and can notify you of consumption and leaks, down to a miniscule leaking, i.e. down to 0.1L/hr.

https://www.alarmgrid.com/products/alarm-com-adc-swm150
https://www.youtube.com/watch?v=T1pCUwWkvu8

My unit is currently connected to my alarm.com security panel (Qolsys IQ2 Panel) via Z-wave and the alarm.com app notifies me of usage(consumption) and water leaks.

Supposedly, the unit has quite a bit of advanced data being generated, that alarm.com does not share to the customer/end user....so, I'm thinking this would be great for open source zwave-js and Home assistant.

How does one suggest to explore such a device and more importantly, since I have one, how can help?

thanks

**Comment by AlCalzone** [comment]
You can try adding it to Z-Wave JS as described in https://suretyhome.com/wp-content/uploads/SWM150_Manual.pdf and go from there. It will likely be shown as an unknown product, so we're going to need the device ID (something like 0x1234-0xabcd-0x9876) shown in the UI to give it a label.

**Comment by Anto79-ops** [comment]
thanks, @AlCalzone 

I can definitely try pairing it with my Z-Wave JS instance in HA. 

Before I do this, it would involve me to exclude it from my current z-wave system (i.e, security system) and that would break my integrations and history there....which is not a problem for me, if this will lead to eventual development for device inclusion into Z-Wave JS.

BUT do I keep i keep connected to Z-wave JS? ie. is just that you want the device ID or did you want me to do other tasks later.

thanks for clearing that up!

**Reply by AlCalzone** [reply]
From reading the manual (and seeing other entries from the same manufacturer), I don't expect to see any configuration parameters. If there are any auto-discovered ones, you can share them too. If there are association groups auto-discovered, those too.
Probably best if you just download a node dump (after the node is ready) or make a [driver log](https://zwave-js.github.io/zwavejs2mqtt/#/troubleshooting/generating-logs?id=driver-logs) that also shows the inclusion, loglevel `debug` or `silly` and attach it here as a file.

Afterwards its really on you to decide. If you want to access some of that additional data, you might want to leave it in Z-Wave JS a bit.

**Comment by Anto79-ops** [comment]
Thanks @AlCalzone for explaining that.

This seems simple enough to do. I was looking at my Zwave  settings, I see all the options to select as in the instructional gif graphic on the link you provided.

I just wanted to make sure I do this right for you.

BEFORE I try to pair the SWM150 to my ZWave system:

1. Do I enable the silly logs to file first, then pair?

2. Once its connected, I'll likely have no functionality, but could I for example:
       a. press the open close button on the unit itself
       b. run water at various flow rates, maybe even cause an alarm

  Would doing the above, add more information to the logs?

3. Since I have other nodes in my ZWave system, will the logs report all nodes? OR should I do a node dump instead of a driver log?

thanks for clearing that up!

**Reply by AlCalzone** [reply]
> Do I enable the silly logs to file first, then pair?

Ideally, yes.

> Once its connected, I'll likely have no functionality

It should have, if there is something to control. Z-Wave functionality is standardized, all we need the configuration files for are labels, predefining config parameters for easy entry and in some cases overriding how the lifeline association is set up.

> Since I have other nodes in my ZWave system, will the logs report all nodes?

It probably will, but I don't care. Better to have everything than to filter out too much.

**Comment by Anto79-ops** [comment]
Just did it, and to my surprise, the parameters are showing up!

This is after interview process:

![image](https://user-images.githubusercontent.com/79593761/184179774-bfd7bc34-1bbf-4c0a-a2a8-4a200f472487.png)

and this is how it looks in HA:

![image](https://user-images.githubusercontent.com/79593761/184179929-981487fa-ae7c-4f3d-ba3e-2a1755ff71cc.png)

Im trying to see if the numbers make sense, but they seem to be updating!

**Comment by Anto79-ops** [comment]
btw, how do I download the log file? I can see it, and check mark it, but not sure how to download it

thanks!

**Comment by Anto79-ops** [comment]
This file, I believe is the node dump as I went to the device, and went to logs.

[zwave_js (2).log](https://github.com/zwave-js/node-zwave-js/files/9310476/zwave_js.2.log)

I can get you the entire log from above (including pairing), once I figure out how to get the file off HA.

**Comment by Anto79-ops** [comment]
ok, nevermind, here is the entire log file, I didn't see the "download" button until I opened the file in z-wave-js!

[zwavejs_2022-08-11.log](https://github.com/zwave-js/node-zwave-js/files/9310524/zwavejs_2022-08-11.log)
.

**Comment by Anto79-ops** [comment]
some updates on the sensors:

I can turn the valve on and off by pressing the first button here:

![image](https://user-images.githubusercontent.com/79593761/184197559-bab23436-8697-4a4e-ac0d-abbce64c2e2a.png)

Not exactly sure what second button does. During an alarm (I have active leak in my home), the volume slider also works to mute the meter.

from these sensors:

![image](https://user-images.githubusercontent.com/79593761/184197780-c97b68dc-ea1c-4133-a090-9f7e9589a5cf.png)

the only thing that I've seen changing is the temperature, and the flow rate, BUT the flow rate I think the units are off, for example, I flush a toilet and it says the rate is like 470L/h...I guess if the water is going for an hour like that, it could be 470L? Im not sure.

Not sure that these do/mean
![image](https://user-images.githubusercontent.com/79593761/184198071-c9628401-a784-40b6-aa97-9ccf905ab42c.png)


the voltage one is an important one, I think that is low flow leak sensor, for example, from the technician point of view, the voltage varies with small leaks and can be used to detect a low flow condition, simulated here.

![image](https://user-images.githubusercontent.com/79593761/184547665-e6166641-7653-44dc-8fd9-b075a134481e.png)

The parameters that can be changed are low flow, medium flow, and high flow conditions, which can be used to make automations such as alerts and even shut the valve off.

EDIT: Remove names form photos.

**Comment by Anto79-ops** [comment]
update, voltage seems to be updating, but not sure if its updating properly (or in time?).

![image](https://user-images.githubusercontent.com/79593761/184218722-0a119580-4504-4eb0-a365-04979b7c4fb2.png)

**Reply by Anto79-ops** [reply]
I just realized I can make my own alerts based on the flow rate:

For example by default a medium flow rate alarm is 1 GPM (or 227L/h) for 40 minutes and high flow rate alarm 7.5 GPM (1705 L/h) for 1 minute.

A low flow rate is considered ANY water movement/flow for 2 hours, and I believe this is where the voltage sensor kicks in because the other sensor is a bit more coarse and is for faster flows. 

Interesting I'm not sure how much of this information is in the device but I do know that the dealer/installer can change these numbers on their end, but this could all be based on just the simple flow rate sensor.

**Comment by Anto79-ops** [comment]
Hi, just some more updates on this.

Yesterday, I was pressing the buttons and what i thought was turning the buzzer off and on, was actually turning the unit off from the network.

![image](https://user-images.githubusercontent.com/79593761/184396667-f97e6c66-da1f-46e8-949e-d864517e3043.png)

I pressed the buzzer button (i.e. second line in the above photo, what looks like a siren?) and the node became instantly unavailable in HA. I then went to the unit and pressed the network  button, and it came back online, but the numbers were not updating.

At this point, I deleted the node, and performed a factory reset on the device, and re-added back.

When I did this, it seems that all the thresholds presets were erased, so it won't alarm anymore on the set thresholds ( I tested a few scenarios for water flow conditions). If you recall, the first time I added the unit to HA, I simply excluded it from the security network, and included it to my HA network. So, it seems it does retain some user/dealer settings that get erased when factory default.

Here how the settings look like from the dealer point of view of the device. I would image this would be required for the device to work properly in zwave-js, otherwise it won't alarm/notify users.

[Water Shut Off and Meter - Setting Options.pdf](https://github.com/zwave-js/node-zwave-js/files/9325135/Water.Shut.Off.and.Meter.-.Setting.Options.pdf)

I suspect that a similar interface should be available in z-wave js for user configuration of temperature and flow thresholds, and calibration.

**Reply by AlCalzone** [reply]
I suspect there are a few undocumented configuration parameters that were used to set the thresholds. It is possible to scan them, but you'll have to use the Z-Wave JS to MQTT addon for that. If you don't have that already, see https://www.home-assistant.io/integrations/zwave_js#can-i-switch-between-the-official-z-wave-js-add-on-and-zwavejs2mqtt and the following section for a description on how to switch without losing entities.

Once inside the addon, make sure [driver logging](https://zwave-js.github.io/zwavejs2mqtt/#/troubleshooting/generating-logs?id=driver-logs) is still enabled. Then you can open the driver function window
![grafik](https://user-images.githubusercontent.com/17641229/184818706-91719cae-8785-43ba-aff6-c3114fc8296a.png)

and execute the following script:
```js
const node = driver.controller.nodes.get(14);
node.commandClasses.Configuration.scanParametersLegacy()
```

You should hopefully see a couple of queries in the log now, I'll need all of them.

**Reply by AlCalzone** [reply]
About the buttons that do something else than intended, maybe they're mis-using Z-Wave features internally and we don't know how. It could be worth opening an issue about that in https://github.com/home-assistant/core/issues, but I'm not sure there's something to be done about that.

**Comment by Anto79-ops** [comment]
Hi, so its been several days now, and a couple of unknown sensors started reporting,

![image](https://user-images.githubusercontent.com/79593761/184705199-bc8cc605-bd0d-4043-a16b-98a3ea7d5397.png)

namely, the water alarm( idle) and water leak alarm (dry).

water consumption and pulse count do not update.

So, @AlCalzone, it seems to work, but not 100%. So do you think some tinkering magic, we could get this device to work, fully with customizable thresholds?

In addition to some of the documents above, found some other documents about the product for more information. thanks

[User-Manual-4505584 (1).pdf](https://github.com/zwave-js/node-zwave-js/files/9339454/User-Manual-4505584.1.pdf)
[adc-swm150 (1).pdf](https://github.com/zwave-js/node-zwave-js/files/9339470/adc-swm150.1.pdf)

**Comment by Anto79-ops** [comment]
thanks, at @AlCalzone 

I think(!) I did the migration to ZWave2mqtt properly...it was weird....I had it already installed even thought I was using Z-wave js so I didn't have to install it...maybe I was already using it...anyways...

I followed your instructions and added two lines (first attempted failed, because I realized I had to add the slashes as shown below)...so you'll see that in the logs

```
//const node = driver.controller.nodes.get(14);
//node.commandClasses.Configuration.scanParametersLegacy()
```
[zwavejs2mqtt_2022-08-16.log](https://github.com/zwave-js/node-zwave-js/files/9353312/zwavejs2mqtt_2022-08-16.log)

**Comment by Anto79-ops** [comment]
hmm, it seems the water meter valve changed from node 14 to node 17, so here's the call for node 17 file.

[zwavejs2mqtt_2022-08-16.log](https://github.com/zwave-js/node-zwave-js/files/9353435/zwavejs2mqtt_2022-08-16.log)


let me know if something doesn't make sense, I suspect I didn't do the migration from z-wave js to z-wave mqtt correctly.

thanks

**Reply by AlCalzone** [reply]
Lines with double slashes are comments, so those two lines you posted in the other comment did exactly nothing. Please try again without double slashes, but with the correct node ID (17).

Also I'm going to need the [driver logs](https://zwave-js.github.io/zwavejs2mqtt/#/troubleshooting/generating-logs?id=driver-logs) you posted the previous times, not the application logs.

Querying the parameters may take a while so give it a good 5 minutes before downloading the log.

**Comment by Anto79-ops** [comment]
Hi @AlCalzone 
thanks for clearing that up!
Ok, I think I got it this time. I let the logs run for about 40 mins after successfully injecting that line of code for node 17 .

Please let me know if something is not right, and I can try again.

[zwavejs_2022-08-17.log](https://github.com/zwave-js/node-zwave-js/files/9363545/zwavejs_2022-08-17.log)

**Comment by Anto79-ops** [comment]
by the way, just for my own information, what does the above do?

Looking at the logs, I see that it is querying parameters from 1 to 255....for example " [Node 017]   trying param 4." Is it just fishing for what the device can perform, and from this a proper driver can be made? thanks!

**Comment by Anto79-ops** [comment]
@AlCalzone Here’s an example the device in action. This device doesn’t really like when we water the grass (never did). My wife was watering the grass, and if the flow is more than 232 L/h for a certain amount of time (40 min), it goes into alarm mode (i.e. medium flow alarm). 
Also, when my wife moved the sprinkler head, she kinked the hose, and you can see that drop in water flow where the arrow is.

As you can see here:

![image](https://user-images.githubusercontent.com/79593761/185639009-f2d4fe3d-885c-4cb7-98b7-8be7ff6f4f90.png)


with the above sensor information, I can create an automation to turn off the valve and send notifications/alerts.

Also the fact that I can see the flow rate live (something not available when using the alarmdotcom platform), I can also set automations based on that number. For example, if the flow rate is more 180 L/h, for more than 10 minutes, create an actionable script to warn us and give options to shut the valve off or turn it off automatically.

Not sure though how to get information for low flow alarms, might have something to do with the voltage sensor (medium and high flow are captured).

EDIT: Also, consumption is not updating. Its stuck at 0.01 since the device was included to the network, about a week ago. Although, looking at the logs, perhaps there some math that needs to be done with these 2 numbers here?

```

2022-08-16 12:11:47.125 INFO ZWAVE: Node 17: value added 17-50-0-value-196609 => 0.01
2022-08-16 12:11:47.126 INFO ZWAVE: Node 17: value added 17-50-0-value-197377 => 0
```

**Comment by AlCalzone** [comment]
Ok this time the scan looked good. It found the following 66 (!) parameters:
```
parameter #: 1
value size:  1
value:       28

parameter #: 2
value size:  1
value:       4

parameter #: 3
value size:  1
value:       70

parameter #: 4
value size:  1
value:       7

parameter #: 5
value size:  1
value:       4

parameter #: 6
value size:  1
value:       4

parameter #: 7
value size:  1
value:       1

parameter #: 8
value size:  1
value:       40

parameter #: 9
value size:  1
value:       2

parameter #: 10
value size:  2
value:       40

parameter #: 11
value size:  2
value:       650

parameter #: 12
value size:  1
value:       4

parameter #: 13
value size:  1
value:       4

parameter #: 14
value size:  1
value:       4

parameter #: 15
value size:  1
value:       10

parameter #: 16
value size:  1
value:       20

parameter #: 17
value size:  1
value:       11

parameter #: 18
value size:  1
value:       2

parameter #: 19
value size:  4
value:       0

parameter #: 20
value size:  1
value:       62

parameter #: 21
value size:  1
value:       78

parameter #: 22
value size:  1
value:       78

parameter #: 23
value size:  1
value:       0

parameter #: 24
value size:  1
value:       10

parameter #: 25
value size:  2
value:       0

parameter #: 26
value size:  2
value:       0

parameter #: 27
value size:  1
value:       5

parameter #: 28
value size:  1
value:       1

parameter #: 29
value size:  1
value:       30

parameter #: 30
value size:  1
value:       3

parameter #: 31
value size:  1
value:       60

parameter #: 32
value size:  1
value:       -106

parameter #: 33
value size:  1
value:       28

parameter #: 34
value size:  1
value:       4

parameter #: 35
value size:  1
value:       70

parameter #: 36
value size:  1
value:       7

parameter #: 37
value size:  1
value:       4

parameter #: 38
value size:  1
value:       4

parameter #: 39
value size:  1
value:       1

parameter #: 40
value size:  1
value:       40

parameter #: 41
value size:  1
value:       2

parameter #: 65
value size:  1
value:       28

parameter #: 66
value size:  1
value:       4

parameter #: 67
value size:  1
value:       70

parameter #: 68
value size:  1
value:       7

parameter #: 69
value size:  1
value:       4

parameter #: 70
value size:  1
value:       4

parameter #: 71
value size:  1
value:       1

parameter #: 72
value size:  1
value:       40

parameter #: 73
value size:  1
value:       2

parameter #: 81
value size:  1
value:       0

parameter #: 82
value size:  1
value:       10

parameter #: 83
value size:  2
value:       60

parameter #: 84
value size:  1
value:       0

parameter #: 85
value size:  1
value:       4

parameter #: 86
value size:  1
value:       8

parameter #: 87
value size:  1
value:       0

parameter #: 88
value size:  1
value:       30

parameter #: 89
value size:  1
value:       0

parameter #: 96
value size:  1
value:       20

parameter #: 97
value size:  1
value:       15

parameter #: 98
value size:  1
value:       3

parameter #: 99
value size:  1
value:       1

parameter #: 100
value size:  1
value:       3

parameter #: 101
value size:  1
value:       15

parameter #: 102
value size:  1
value:       1
```

I'm afraid though that this will require quite a bit of manual work to figure out what they are (cross-referencing with the configuration UI you've showed before) and which limits they have (try and error :( ).

It will likely be easier asking the manufacturer for an overview of the configuration parameters in the form of a manual.

**Reply by Anto79-ops** [reply]
@AlCalzone ahh, ok. I can ask some people, but won't hold my breath for this. Is such information proprietary? Is so, there would be no way for me to get that :(

**Comment by AlCalzone** [comment, marked as answer]
https://github.com/zwave-js/node-zwave-js/pull/5557
